pkg/chains/chain.go

```diff
@@ -87,99 +76,68 @@ func (chain Chain) EncodeAddress(b []byte) (string, error) {
 	return "", fmt.Errorf("chain (%d) not supported", chain.ChainId)
 }
 
-func (chain Chain) BTCAddressFromWitnessProgram(witnessProgram []byte) (string, error) {
-	chainParams, err := GetBTCChainParams(chain.ChainId)
-	if err != nil {
-		return "", err
-	}
-	address, err := btcutil.NewAddressWitnessPubKeyHash(witnessProgram, chainParams)
-	if err != nil {
-		return "", err
-	}
-	return address.EncodeAddress(), nil
-}
-
-// DecodeAddress decode the address string to bytes
-func (chain Chain) DecodeAddress(addr string) ([]byte, error) {
-	return DecodeAddressFromChainID(chain.ChainId, addr)
-}
-
 // DecodeAddressFromChainID decode the address string to bytes
-func DecodeAddressFromChainID(chainID int64, addr string) ([]byte, error) {
-	if IsEVMChain(chainID) {
+// additionalChains is a list of additional chains to search from
+// in practice, it is used in the protocol to dynamically support new chains without doing an upgrade
+func DecodeAddressFromChainID(chainID int64, addr string, additionalChains []Chain) ([]byte, error) {
```

Comment:
Same, the parameter name `additionalChains` could confuse devs. The only names I could think of was `chainBook` or `chainDictionary`